app/models/manageiq/providers/nutanix/inventory/collector/target_collection.rb

```diff
@@ -0,0 +1,85 @@
+class ManageIQ::Providers::Nutanix::Inventory::Collector::TargetCollection < ManageIQ::Providers::Nutanix::Inventory::Collector
+  def initialize(_manager, target)
+    super
+    parse_targets!
+    infer_related_ems_refs!
+
+    # Reset the target cache, so we can access new targets inside
+    target.manager_refs_by_association_reset
+  end
+
+  def clusters
+    return [] if references(:ems_clusters).blank?
+    return @clusters if @clusters
+
+    clusters_api = NutanixClustermgmt::ClustersApi.new(cluster_mgmt_connection)
+    @clusters = references(:ems_clusters).map do |cluster_ref|
+      clusters_api.get_cluster_by_id(cluster_ref).data
+    end
+  end
+
+  def hosts
+    return [] if references(:hosts).blank?
+    return @hosts if @hosts
+
+    clusters_api = NutanixClustermgmt::ClustersApi.new(cluster_mgmt_connection)
+    @hosts = references(:hosts).map do |host_ref|
+      clusters_api.get_host_by_id(host_ref).data
```

Comment:
TODO I have to add in the cluster_ref here as well
```suggestion
      clusters_api.get_host_by_id(cluster_ref, host_ref).data
```